### ISSUE TITLE
Allow replicator documents to include params for db creation (q)

### ISSUE THREAD

**Issue by wohali** [apache/couchdb#887]
If you specify `"create_database": true` in a replicator doc, the replicator creates the target database, with the default q value.

It'd be real neat if you could also specify `"create_database_params": "q=1"` to have the replicator request the target to create the database with `q=1`. Right now, if you want this behaviour, you have to have a client create the target DB itself, then start the replication up without the `create_database` option.

@janl and I see a number of situations where this could be really helpful in the field.

**Comment by jiangphcn** [comment]
@wohali do you mean `"create_target": true` instead of `"create_database": true` in replicator doc?
If we introduce `"create_database_params": "q=1"`, what about using syntax like `"create_database_params": "[ {ParamName, ParamValue}, {ParamName, ParamValue}, ... ]"`, such as `"create_database_params": "[ {q, 1}]"`?

**Comment by nickva** [comment]
@jiangphcn this would live in the replicator document body so it would look more like json fields and values `"create_target_params": {"q": "1", ...}`

**Comment by jiangphcn** [comment]
@nickva yes, make sense. Let me add this support if don't mind :-)

**Comment by nickva** [comment]
@jiangphcn sounds great, thanks for giving it a try!

**Comment by jiangphcn** [comment]
https://github.com/apache/couchdb/pull/1017 was merged.

**Comment by eiri** [comment]
Closed by #1017

**Comment by vincnetas** [comment]
I think this parameter was left undocumented. Could update  documentation but not sure what was added and what's the syntax. Can't read erlang :( 

Is it root level property :

```
{
  "create_target": true,
  "create_target_params": {"q": "1", ...}
}
```
